### ISSUE TITLE
lib2to3.pgen2.parse.ParseError: bad input: type=8, value=u')',

### ISSUE THREAD

**Issue by cancan101** [google/yapf#412]
`yapf -d err.py` dies with (contents of err.py):
```py
args = tuple(x for x in [2],)
```

this is valid python 2.7.

```py-tb
Traceback (most recent call last):
  File "/Users/alex/.virtualenvs/yapf/bin/yapf", line 11, in <module>
    sys.exit(run_main())
  File "/Users/alex/.virtualenvs/yapf/lib/python2.7/site-packages/yapf/__init__.py", line 296, in run_main
    sys.exit(main(sys.argv))
  File "/Users/alex/.virtualenvs/yapf/lib/python2.7/site-packages/yapf/__init__.py", line 188, in main
    parallel=args.parallel)
  File "/Users/alex/.virtualenvs/yapf/lib/python2.7/site-packages/yapf/__init__.py", line 236, in FormatFiles
    in_place, print_diff, verify)
  File "/Users/alex/.virtualenvs/yapf/lib/python2.7/site-packages/yapf/__init__.py", line 259, in _FormatFile
    logger=logging.warning)
  File "/Users/alex/.virtualenvs/yapf/lib/python2.7/site-packages/yapf/yapflib/yapf_api.py", line 90, in FormatFile
    verify=verify)
  File "/Users/alex/.virtualenvs/yapf/lib/python2.7/site-packages/yapf/yapflib/yapf_api.py", line 126, in FormatCode
    tree = pytree_utils.ParseCodeToTree(unformatted_source)
  File "/Users/alex/.virtualenvs/yapf/lib/python2.7/site-packages/yapf/yapflib/pytree_utils.py", line 108, in ParseCodeToTree
    tree = parser_driver.parse_string(code, debug=False)
  File "/Users/alex/.pyenv/versions/2.7.12/Python.framework/Versions/2.7/lib/python2.7/lib2to3/pgen2/driver.py", line 106, in parse_string
    return self.parse_tokens(tokens, debug)
  File "/Users/alex/.pyenv/versions/2.7.12/Python.framework/Versions/2.7/lib/python2.7/lib2to3/pgen2/driver.py", line 71, in parse_tokens
    if p.addtoken(type, value, (prefix, start)):
  File "/Users/alex/.pyenv/versions/2.7.12/Python.framework/Versions/2.7/lib/python2.7/lib2to3/pgen2/parse.py", line 159, in addtoken
    raise ParseError("bad input", type, value, context)
lib2to3.pgen2.parse.ParseError: bad input: type=8, value=u')', context=('', (1, 28))
```

**Comment by taion** [comment]
That's valid Python period, BTW – in Python 2 or Python 3.

**Comment by cancan101** [comment]
This looks to be the upstream issue: https://bugs.python.org/issue27494

**Comment by gwelymernans** [comment]
We're (fortunately or unfortunately) tethered to lib2to3's parsing. It's why some of the Python3 issues haven't been addressed yet. This appears to be another issue we need to wait on them for...
